moma_controllers/path_admittance_controller/src/path_admittance_controller.cpp

```diff
@@ -135,41 +136,39 @@ void PathAdmittanceController::update(const ros::Time& time, const ros::Duration
   Eigen::Matrix3d R(q);
 
   // Update measured wrench and tracking errors
-  force_ext_ = Eigen::Vector3d(wrench_.wrench.force.x,
-                                 wrench_.wrench.force.y, 
-                                 wrench_.wrench.force.z);
+  force_ext_ = Eigen::Vector3d(!wrench_inverted_ ? wrench_.wrench.force.x : -wrench_.wrench.force.x,
+                               !wrench_inverted_ ? wrench_.wrench.force.y : -wrench_.wrench.force.y,
+                               !wrench_inverted_ ? wrench_.wrench.force.z : -wrench_.wrench.force.z);
   threshold(force_ext_, force_threshold_);
 
-  force_integral_ += force_ext_ * dt;
-  force_integral_ = force_integral_.cwiseMax(-force_integral_max_);
-  force_integral_ = force_integral_.cwiseMin(force_integral_max_);
-  Eigen::Vector3d delta_position =
-      Kp_linear_.cwiseProduct(force_ext_) + Ki_linear_.cwiseProduct(force_integral_);
-  Eigen::Vector3d delta_position_transformed = R * delta_position;
+  dpos_ddot_ = /*M_inv_ **/ (force_ext_ - Kd_linear_.cwiseProduct(dpos_dot_) - Kp_linear_.cwiseProduct(dpos_));
```

Comment:
Rather then commenting, keep an M_inv_ matrix somewhere which is the diagonal identity matrix.